audmetric/core/utils.py

```diff
@@ -20,10 +20,24 @@ def infer_labels(
         prediction: typing.Sequence[typing.Any],
         labels: typing.Optional[typing.Sequence[typing.Any]],
 ) -> typing.Sequence[typing.Any]:
-    r"""Infer labels from truth and prediction."""
+    r"""Infer labels from truth and prediction.
+
+    It gathers all labels that are present in the truth and prediction values.
+
+    Args:
+        truth: ground truth labels
+        prediction: predicted labels
+        labels: if labels are provided,
+            they will not be infered
+            but simply returned as a sorted list
+
+    Returns:
+        labels
+
+    """
     if labels is None:
-        labels = sorted(list(set(truth) | set(prediction)))
-    return labels
+        labels = set(truth) | set(prediction)
+    return sorted(list(labels))
```

Comment:
Are we sure this cannot have unwanted side-effects? Before we were simply returning the labels in the original order.

Maybe it makes more sense to have two functions:

* public function takes `truth` and `prediction` but not `labels` and infers the labels
* internal function takes `truth`, `prediction` and `labels` and calls the public function if `labels` are not given or otherwise simply returns them

Reply:
Yes, makes sense.

The `labels` argument was originally one of the reasons why I decided to make `infer_labels()` public, as I thought this is a little bit strange to have.

So, I agree that we should remove `labels` here as argument and use it only internally.

Reply:
I removed the labels argument and updated the screenshot in the description accordingly.

Reply:
great, thanks